### DISCUSSION TITLE
Is there a way to provide a default value for a newly created required field to old documents?

### DISCUSSION THREAD

**Discussion by bn-piar** [prisma/prisma#17995] [Q&A] [answered]
Hi. I'm using Mongodb. 

I often add new fields which by design are required but if I make them required it breaks my web-app because old documents don't have this field. 

To avoid this I'm making these fields optional. 

Is there a way to populate old documents with a default value? I mean without writing extra logic

Tried to push new schema to the db but it doesn't populate old documents with default values for the field.

**Comment by nurul3101** [comment, marked as answer]
Hey @bn-piar 👋 

I would recommend you to write a migration script that populates the old documents with a default value. 

If you would mark a field having a default value in schema file, only the new documents that are inserted would be populated with the default value, you would still need to write a script to add default values to already inserted documents.

Please let me know in case you need any assistance in coming up with a migration script.

**Reply by bn-piar** [reply]
I'd love a simple example of migration script if it's not too bothersome =)

**Reply by nurul3101** [reply]
Sure, Can you share your models which needs to be migrated?

**Reply by rono-koushiq** [reply]
Hello, I am facing this same issue. Can you please help me ? the model is given below -

model build {
  id              String       @id @default(auto()) @map("_id") @db.ObjectId
  buildId         String
  buildDir        String       @unique
  uploadRetry     Int?         @default(0)
}

uploadRetry is the newly created field which doesnt exist for the previous records

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
